Source/OIDAuthState.m

```diff
@@ -472,40 +472,74 @@ - (void)performActionWithFreshTokens:(OIDAuthStateAction)action
   // refresh the tokens
   OIDTokenRequest *tokenRefreshRequest =
       [self tokenRefreshRequestWithAdditionalParameters:additionalParameters];
+  __weak OIDAuthState *weakSelf = self;
   [OIDAuthorizationService performTokenRequest:tokenRefreshRequest
                  originalAuthorizationResponse:_lastAuthorizationResponse
                                       callback:^(OIDTokenResponse *_Nullable response,
                                                  NSError *_Nullable error) {
     dispatch_async(dispatch_get_main_queue(), ^() {
+      __strong OIDAuthState *strongSelf = weakSelf;
       // update OIDAuthState based on response
       if (response) {
-        self->_needsTokenRefresh = NO;
-        [self updateWithTokenResponse:response error:nil];
+        strongSelf->_needsTokenRefresh = NO;
```

Comment:
If a weak reference is really required here instead of capturing a strong reference to self, then the implication there is that strongSelf may be nil here, but there is no check for that condition. Dereferencing nil on this line will crash the application.

I am not a fan of applying this weak reference pattern for every single block - though I know some people are.

It's my view that; in the case where a block is guaranteed to be invoked and released (vs a indefinitely-lived block like the kind you give NSNotificationCenter) there is nothing wrong (and arguably it's more correct) to have a strong reference captured by the block. Afterall, we don't solve the problem of possible retain cycles by simply making every object reference in our code weak... we want some references to be strong references.

AFAICT, This call should either succeed or fail, and will always invoke this block in either case, and the block should not be retained beyond that point. WDYT?